src/ksm/client/distributed_pod_label_discovery.go

```diff
@@ -36,7 +35,7 @@ func (p *distributedPodLabelDiscoverer) findAllLabeledPodsRunningOnNode() ([]v1.
 		}
 
 		if pod.Status.HostIP == p.ownNodeIP {
-			p.logger.Debugf("Found KSM pod running on this code, pod IP: %s", pod.Status.PodIP)
+			p.logger.Debugf("Found KSM pod running on this node, pod IP: %s", pod.Status.PodIP)
 			foundPods = append(foundPods, pod)
 		}
```

Comment:
Maybe a bit unrelated, but do you think it would be worth logging (`Debugf`) the pods that do not match this condition? Could be helpful for debugging.

Reply:
If you query all namespaces, it will be a lot. Maybe we can log a pod selector in namespace X + IP which we look for?

In case all pods on the cluster are queried (this is the default), it will be a lot of output.

Reply:
I was under the impression that this check is evaluated after filtering by the KSM label, isn't this the case?

Reply:
Yes, it is. I'll see how it would look like.